pkg/network/dhcpconfigurator/dhcp_configurator.go

```diff
@@ -0,0 +1,79 @@
+)
+
+type DhcpConfigurator struct {
+	advertisingIfaceName     string
+	dhcpConfigurationFactory cache.InterfaceCacheFactory
+	filterByMac              bool
+	handler                  netdriver.NetworkHandler
+	launcherPID              string
+}
+
+// NewDhcpConfiguratorWithClientFilter should be used when the DHCP server is
+// expected to only reply to the MAC specified in the `cache.DhcpConfig` struct
+func NewDhcpConfiguratorWithClientFilter(configurationFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) DhcpConfigurator {
+	return DhcpConfigurator{
+		advertisingIfaceName:     advertisingIfaceName,
+		dhcpConfigurationFactory: configurationFactory,
+		launcherPID:              launcherPID,
+		filterByMac:              true,
+		handler:                  handler,
+	}
+}
+
+// NewDhcpConfigurator should be used when the DHCP server is
+// expected to reply all client requests, independently of their MAC address
+func NewDhcpConfigurator(configurationFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) DhcpConfigurator {
+	return DhcpConfigurator{
+		advertisingIfaceName:     advertisingIfaceName,
+		dhcpConfigurationFactory: configurationFactory,
+		launcherPID:              launcherPID,
+		filterByMac:              false,
+		handler:                  handler,
+	}
+}
+
+func (d DhcpConfigurator) ImportConfiguration(ifaceName string) (*cache.DhcpConfig, error) {
+	dhcpConfig, err := d.dhcpConfigurationFactory.CacheDhcpConfigForPid(d.launcherPID).Read(ifaceName)
+	if err != nil {
+		return nil, err
+	}
+	dhcpConfig.AdvertisingIPAddr = dhcpConfig.AdvertisingIPAddr.To4()
+	dhcpConfig.AdvertisingIPv6Addr = dhcpConfig.AdvertisingIPv6Addr.To16()
+	return dhcpConfig, nil
+}
+
+func (d DhcpConfigurator) ExportConfiguration(config cache.DhcpConfig) error {
+	return d.dhcpConfigurationFactory.CacheDhcpConfigForPid(d.launcherPID).Write(config.Name, &config)
+}
+
+func (d DhcpConfigurator) EnsureDhcp(podInterfaceName string, dhcpConfig cache.DhcpConfig, dhcpOptions *v1.DHCPOptions) error {
+	dhcpStartedFile := fmt.Sprintf("/var/run/kubevirt-private/dhcp_started-%s", podInterfaceName)
+	_, err := os.Stat(dhcpStartedFile)
+	if os.IsNotExist(err) {
+		if err := d.startServer(dhcpConfig, d.advertisingIfaceName, dhcpOptions); err != nil {
+			return fmt.Errorf("failed to start DHCP server for interface %s", podInterfaceName)
+		}
+		newFile, err := os.Create(dhcpStartedFile)
+		if err != nil {
+			return fmt.Errorf("failed to create dhcp started file %s: %s", dhcpStartedFile, err)
+		}
+		newFile.Close()
+	}
+	return nil
+}
+
+func (d DhcpConfigurator) startServer(config cache.DhcpConfig, advertisingIfaceName string, dhcpOptions *v1.DHCPOptions) error {
+	if !config.IPAMDisabled {
+		return d.handler.StartDHCP(&config, config.AdvertisingIPAddr, advertisingIfaceName, dhcpOptions, d.filterByMac)
+	}
+	return nil
+}
```

Comment:
Let's unify that into a `Start`

Reply:
Let's not; that'll complicate the unit tests. 

This keeps the current behavior, which allows me to unit test starting the server, and not caring about the file that is used to know if the dhcp server is started.